transmission_interface/src/transmission_parser.cpp

```diff
@@ -12,68 +12,250 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "transmission_interface/transmission_parser.hpp"
-#include "transmission_interface/transmission_info.hpp"
+#include <transmission_interface/transmission_parser.hpp>
 
-#include <tinyxml2.h>
-#include <stdexcept>
+// ROS
+#include <rclcpp/rclcpp.hpp>
+
+#include <sstream>
 #include <string>
 #include <vector>
 
-namespace
-{
-constexpr const auto kTransmissionTag = "transmission";
-constexpr const auto kJointTag = "joint";
-constexpr const auto kHardwareInterfaceTag = "hardwareInterface";
-}  // namespace
 
 namespace transmission_interface
 {
+static const rclcpp::Logger g_logger(rclcpp::get_logger("transmission_parser"));
```

Comment:
We are actually using something like this throughout the framework:
 ```suggestion
namespace
{
constexpr const auto kLoggerName = "ros2_control_resource_manager";
}  // namespace
```

with `rclcpp::get_logger(kLoggerName)`.